### ISSUE TITLE
async/await doesn't work with chained Deferreds

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8890]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8890|
|Type|enhancement|
|Created|2016-11-03 00:06:29Z|
|Branch|https://github.com/twisted/twisted/tree/8890-chained|

If you have a Deferred waiting on another Deferred, async/await will yield that (unresolved) other deferred instead of the result it will eventually return.

Example:

```
from twisted.internet.defer import ensureDeferred
from twisted.internet.error import DNSLookupError
from twisted.names.client import getHostByName
from twisted.names.error import DNSNameError
from twisted.internet.task import react

async def hostname(name):
    try:
        d = getHostByName(name)
        address = await d
    except DNSNameError:
        return "no such host"
    except DNSLookupError:
        return "lookup error"

    return address


if __name__ == "__main__":
    react(ensureDeferred(hostname("google.com")))
```

<details><summary>Searchable metadata</summary>

```
trac-id__8890 8890
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Twisted_16_6 Twisted 16.6
branch__8890_chained 8890-chained
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1478131589481732 1478131589481732
changetime__1484655683433640 1484655683433640
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

PR at https://github.com/twisted/twisted/pull/584

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez set owner to @hawkowl|
|-|-|

reviewed, tested with my klein app and fixed my issue there.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 837cd70da622f82701eddfe7cd15b0b799988070
```
#!CommitTicketReference repository="" revision="837cd70da622f82701eddfe7cd15b0b799988070"
Merge 8890-chained: Fix awaiting chained/paused Deferreds

Author: hawkowl
Reviewer: wsanchez
Fixes: #8890
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc commented|
|-|-|

In changeset 3257fbb3504329f603cc9b6e17442d913170b5d1
```
#!CommitTicketReference repository="" revision="3257fbb3504329f603cc9b6e17442d913170b5d1"
Merge pull request #682 from twisted/8890-openssh-docstring-2

Author: Julian
Reviewer: rodrigc
Fixes: ticket:8890

Remove an old OpenSSH reference
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc set status to `reopened`|
|-|-|

Closed wrong ticket.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

It looks to me like this was closed before.
